src/wp-admin/includes/image.php

```diff
@@ -301,18 +308,36 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 			if ( ! is_wp_error( $resized ) ) {
 				// Append "-scaled" to the image file name. It will look like "my_image-scaled.jpg".
 				// This doesn't affect the sub-sizes names as they are generated from the original image (for best quality).
-				$saved = $editor->save( $editor->generate_filename( 'scaled' ) );
+				$saved = $editor->save( $editor->generate_filename( 'scaled' ), $primary_mime_type );
 
 				if ( ! is_wp_error( $saved ) ) {
 					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
 
+					$image_meta['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $saved );
+
 					// If the image was rotated update the stored EXIF data.
 					if ( true === $rotated && ! empty( $image_meta['image_meta']['orientation'] ) ) {
 						$image_meta['image_meta']['orientation'] = 1;
 					}
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
 				} else {
 					// TODO: Log errors.
 				}
+
+				// Save additional mime types.
+				foreach ( $additional_mime_types as $output_mime_type ) {
+					if ( empty( $mime_extension_map[ $output_mime_type ] ) ) {
+						// Skip mime types that are not supported.
+						continue;
+					}
+
+					$extension = explode( '|', $mime_extension_map[ $output_mime_type ] )[0];
+					$saved     = $editor->save( $editor->generate_filename( 'scaled', null, $extension ), $output_mime_type );
```

Comment:
when I tried $editor->save( $editor->generate_filename( null, null, $extension ), $output_mime_type ); the filename became "file-scaled-2560x1920.webp" - 
i tried using `$editor->save( $editor->generate_filename( 'scaled' ), $output_mime_type );` then the filename was `file-scaled-scaled.webp`. this may be a reason I was resetting the editor, to get the right filename output. Open to suggestions for fixing this issue, otherwise the best i can do here would be `file-webp-scaled.webp`, or we could alter the internals further.

Reply:
this is resolved using the new approach of one editor per mime type